internal/nginx/config/validation/common.go

```diff
@@ -46,3 +48,145 @@ func validateEscapedStringNoVarExpansion(value string, examples []string) error
 	}
 	return nil
 }
+
+const invalidHostHeaderErrMsg string = "redefining the Host request header is not supported"
+
+func validateHeaderName(name string) error {
```

Comment:
while this validation might seem restrictive -- it doesn't match  the Gateway API validation rules `+kubebuilder:validation:Pattern=^[A-Za-z0-9!#$%&'*+\-.^_\x60|~]+$`  -- it is correct for NGINX. 

NGINX will validate request headers using this function https://github.com/nginx/nginx/blob/6915d2fb2e88e0c339fe37b37ce14f5fe446c1c6/src/http/ngx_http_parse.c#L819. If it contains something invalid, it will reject the header printing a message in the log. For example:
```
client sent invalid header line: "h]h: hello" while reading client request headers
```

Reply:
I dug into this deeper, and the k8svalidation.IsHTTPHeaderName function is more actually restrictive than the NGINX validation (alphanumeric characters and hyphens only), so for consistency with the API, I think we should keep the more restrictive function. I added a couple of additional tests catching the `&[]` special characters.